sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala

```diff
@@ -202,7 +202,7 @@ class CliSuite extends SparkFunSuite {
            |Exception: $cause
            |Failed to capture next expected output "${expectedAnswers(next)}" within $timeout.
            |
-           |${buffer.mkString("\n")}
+           |${buffer.toSeq.mkString("\n")}
```

Comment:
But the process of `.toSeq` still involves iteration, what is the difference between this iteration and the iteration of `.mkString`?

Reply:
You are right..

Reply:
Should we switch to a thread-safe collection like java.util.concurrent.ConcurrentLinkedQueue or use `synchronized`?

Reply:
I'd like to use the existing lock to synchronize both the buffer and next